client/allocrunner/alloc_runner.go

```diff
@@ -909,6 +909,7 @@ func (ar *allocRunner) SetNetworkStatus(s *structs.AllocNetworkStatus) {
 	ar.stateLock.Lock()
 	defer ar.stateLock.Unlock()
 	ar.state.NetworkStatus = s.Copy()
+	ar.hookResources.SetAllocNetworkStatus(s) // will copy in getter
```

Comment:
The `ar.state.NetworkStatus` has to copy because it's going to persist it to disk. The `SetAllocNetworkStatus` is just keeping it around in memory. I was thinking I'd rather not copy that needlessly if we hit a code path where we're not going to use it. But I think it's safe for us to use the same object we've copied and persisted to disk, because we're encoding it before writing it to the boltdb anyways. Will fix.

Reply:
Done. I've left the extra copy on the hook resource getter, because not doing so is asking for us to always make sure we're making a copy if we end up with an independent caller of `SetAllocNetworkStatus`.